packages/utils/src/libp2p/index.ts

```diff
@@ -16,6 +16,36 @@ export function selectRandomPeer(peers: Peer[]): Peer | undefined {
   return peers[index];
 }
 
+/**
+ * Returns the peer with the lowest latency.
+ * @param getPing - A function that returns the latency for a given peer
+ * @param peers - The list of peers to choose from
+ * @returns The peer with the lowest latency, or undefined if no peer could be reached
+ */
+export async function selectLowestLatencyPeer(
+  getPing: (peerId: PeerId) => Promise<number>,
```

Comment:
The keep alive manager does the pinging already, no need to pass the reference to the protocol, just use the map. if no value is available, then fallback by selecting a random peer.